models/icepack/model_mod.f90

```diff
@@ -0,0 +1,885 @@
+! is not present) This way, we are decreasing the number of distance
+! computations that will follow.  This is a horizontal-distance operation and
+! we don't need to have the relevant vertical coordinate information yet
+! (for obs).
+call loc_get_close_state(filt_gc, base_loc, base_type, locs, loc_qtys, loc_indx, &
+                         num_close, close_indices)
+
+! Loop over potentially close subset of obs priors or state variables
+if (present(distances)) then
+   do k = 1, num_close
+
+      t_ind = close_indices(k)
+
+      ! if dry land, leave original 1e9 value.  otherwise, compute real dist.
+      distances(k) = get_dist(base_loc,      locs(t_ind), &
+                                 base_type, loc_qtys(t_ind))
+   enddo
+endif
+
+end subroutine get_close_state
+
+!------------------------------------------------------------------
+
+function read_model_time(filename)
+
+character(len=256) :: filename
+type(time_type) :: read_model_time
+
+integer :: ncid         ! netcdf file id
+integer :: nyr      , & ! year number, in cice restart
+           month    , & ! month number, 1 to 12, in cice restart
+           mday     , & ! day of the month, in cice restart
+           sec          ! elapsed seconds into date, in cice restart
+integer :: hour     , & ! hour of the day, needed for dart set_date
+           minute   , & ! minute of the hour, needed for dart set_date
+           secthismin
+
+if ( .not. module_initialized ) call static_init_model
+
+if ( .not. file_exist(filename) ) then
+   write(string1,*) 'cannot open file ', trim(filename),' for reading.'
+   call error_handler(E_ERR, 'read_model_time', string1, source)
+endif
+
+call nc_check( nf90_open(trim(filename), NF90_NOWRITE, ncid), &
+                  'read_model_time', 'open '//trim(filename))
+call nc_check( nf90_get_att(ncid, NF90_GLOBAL, 'nyr'  , nyr), &
+                  'read_model_time', 'get_att nyr')
+call nc_check( nf90_get_att(ncid, NF90_GLOBAL, 'month' , month), &
+                  'read_model_time', 'get_att month')
+call nc_check( nf90_get_att(ncid, NF90_GLOBAL, 'mday'   , mday), &
+                  'read_model_time', 'get_att mday')
+call nc_check( nf90_get_att(ncid, NF90_GLOBAL, 'sec', sec), &
+                  'read_model_time', 'get_att sec')
+
+if (nyr == 0) then
+  call error_handler(E_ERR, 'read_model_time', &
+                     'A model time with year 0 is not supported', & 
+                     source)
+endif
+
+hour       = int(sec/3600)
+minute     = int((sec-hour*3600)/60)
+secthismin = int(sec-hour*3600-minute*60)
+
+read_model_time = set_date(nyr, month, mday, hour, minute, secthismin)
+
+end function read_model_time
+
+!------------------------------------------------------------------
+
+subroutine write_model_time(ncid, model_time, adv_to_time)
+
+integer,         intent(in)           :: ncid
+type(time_type), intent(in)           :: model_time
+type(time_type), intent(in), optional :: adv_to_time
```

Comment:
Is write_model_time called the adv_to_time by any icepack code?

' adv_to_time' is not a required argument to write_model_time, so you can remove the argument rather than checking and the erroring out.

Reply:
write_model_time is not called by any of the icepack code

Need to look into using already written versions of this subroutine instead of adding them to the model_mod. There is a write_model_time in the dart_time_io_mod

adv_to_time will be removed